lib/geo.js

```diff
@@ -12,19 +12,31 @@ var assert = require('assert');
 exports.nearFilter = function nearFilter(where) {
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
+  var keyPath = [];
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      var keys = Object.keys(node);
+
+      for (var i = 0; i < keys.length; i++) {
+        var ex = node[keys[i]];
+        keyPath.push(keys[i]);
+
+        if (ex && ex.near) {
```

Comment:
@etsuo According to the logic here, the loop always return either false or recursive search when the first key checked, but I think he intended to go over all keys to see if near exist right?

I am trying to understand the logic in your implementation, correct me if I am wrong lol, thanks.